packages/mdx/mdx-hast-to-jsx.js

```diff
@@ -111,23 +111,20 @@ function toJSX(node, parentNode = {}, options = {}) {
       `const layoutProps = {
   ${exportNames.join(',\n')}
 };
-${skipExport ? '' : 'export default'} class MDXContent extends React.Component {
-  constructor(props) {
-    super(props)
-    this.layout = ${layout || 'null'}
-  }
-  render() {
-    const { components, ...props } = this.props
-    const Layout = this.layout
-
-    return <div
-             name="wrapper"
-             components={components}>
-             ${layout ? `<Layout {...layoutProps} {...props}>` : ''}
-             ${jsxNodes.map(childNode => toJSX(childNode, node)).join('')}
-             ${layout ? `</Layout>` : ''}
-           </div>
-  }
+${
+  skipExport ? '' : 'export default'
+} function MDXContent({ components, ...props }) {
+  const Layout = ${layout || 'null'}
```

Comment:
I think we're going to need to place the layout outside of the function because it will be unstable if the default export is an inlined function:

```md
# Hello, world!

The following function will result in an unstable component:

export default props => <main {...props} />
```

I wonder if we need to do something like:

```js
const MDXLayout = ${layout || 'null'}

// ...

<div
  name="wrapper"
  components={components}>
  <MDXLayout {...layoutProps} {...props}>
     {jsxNodes}
  </MDXLayout>
</div>
```

Reply:
Definitely! No reason to initialize it on each render, it should be defined outside.

Reply:
Done! Can change the name to `MDXLayout` if you'd like as well, might be a bit clearer.